File: TripKitUI/controller/TKUISheetViewController.m
//
//  TimeViewController.m
//  TripGo
//
//  Created by Adrian Schoenig on 24/09/13.
//
//

#import "TKUISheetViewController.h"

#import "TKUISheet.h"

@interface TKUISheetViewController ()

@end

@implementation TKUISheetViewController

- (instancetype)initWithSheet:(TKUISheet *)sheet
{
  self = [super init];
  if (self) {
    self.view = sheet;
  }
  return self;
}

- (TKUISheet *)sheet
{
  if ([self.view isKindOfClass:[TKUISheet class]])
    return (TKUISheet *) self.view;
  else
    return nil;
}

#pragma mark - UIViewController

- (void)viewDidDisappear:(BOOL)animated
{
  [super viewDidDisappear:animated];
  
  [self.sheet doneButtonPressed:nil];
}

- (CGSize)preferredContentSize
{
	return self.sheet.frame.size;
}


@end

File: TripKitUI/controller/TKUISheetViewController.swift
//
//  TKUISheetViewController.swift
//  TripKitUI-iOS
//
//  Created by Adrian Schönig on 29.10.18.
//  Copyright © 2018 SkedGo Pty Ltd. All rights reserved.
//

import UIKit

@objc
public class TKUISheetViewController: UIViewController {
  
  @objc
  public init(sheet: TKUISheet) {
    super.init(nibName: nil, bundle: nil)
    self.view = sheet
  }
  
  required init?(coder aDecoder: NSCoder) {
    fatalError("init(coder:) has not been implemented")
  }
  
  public var sheet: TKUISheet? {
    return self.view as? TKUISheet
  }
  
  public override func viewDidDisappear(_ animated: Bool) {
    super.viewDidDisappear(animated)
    
    sheet?.doneButtonPressed(nil)
  }
  
  public override var preferredContentSize: CGSize {
    get {
      return sheet?.bounds.size ?? .zero
    }
    set {
      // do nothing
    }
  }
  
}
